### PR TITLE
stack command

### PR DESCRIPTION
**Pull request by aarzilli**
I've attempted an implementation of the stacktrace command for issues #64 and #63.

### CODE REVIEW

**Comment by derekparker** [comment]
At a glance this looks good, and everything worked when I pulled it down.

I'll have more time for a proper review this afternoon, however in the meantime I did notice some commented out code and a few `println`'s laying around.

File: terminal/command.go [hunk 1]
```diff
@@ -392,7 +393,7 @@ func info(client service.Client, args ...string) error {
 		data = filterVariables(vars, filter)
 
 	default:
-		return fmt.Errorf("unsupported info type, must be args, funcs, locals, sources, or vars")
+		return fmt.Errorf("unsupported info type, must be args, funcs, locals, sources, goroutines or vars")
```
Comment (derekparker) [inline terminal/command.go]: I don't understand this change, `info` does not have a `goroutines` subcommand.

File: proc/proc_test.go [hunk 2]
```diff
@@ -414,3 +414,114 @@ func TestSwitchThread(t *testing.T) {
...
+
+			if stackMatch(mainStack, locations) {
+				mainCount++
+			}
+
+			if stackMatch(agoroutineStack, locations) {
+				agoroutineCount++
+			}
+
+			/*
```
Comment (derekparker) [inline proc/proc_test.go]: Remove commented code.

File: proc/proc_test.go [hunk 3]
```diff
@@ -414,3 +414,114 @@ func TestSwitchThread(t *testing.T) {
...
+		assertNoError(p.Continue(), t, "Continue()")
+
+		gs, err := p.GoroutinesInfo()
+		assertNoError(err, t, "GoroutinesInfo")
+
+		agoroutineCount := 0
+		mainCount := 0
+
+		for _, g := range gs {
+			//println("Goroutine", g.Id)
```
Comment (derekparker) [inline proc/proc_test.go]: Remove this line.

File: terminal/command.go [hunk 4]
```diff
@@ -404,6 +405,45 @@ func info(client service.Client, args ...string) error {
...
+	stack, err := client.Stacktrace(goroutineid, depth)
+	if err != nil {
+		return err
+	}
+	for i := range stack {
+		name := "(nil)"
+		if stack[i].Function != nil {
+			name = stack[i].Function.Name
+		}
+		fmt.Printf("%d. %s\n\t%s:%d (%d)\n", i, name, stack[i].File, stack[i].Line, stack[i].PC)
```
Comment (derekparker) [inline terminal/command.go]: Change to: (%#v) for printing PC.

File: terminal/command.go [hunk 5]
```diff
@@ -189,7 +190,7 @@ func goroutines(client service.Client, args ...string) error {
 		if g.Function != nil {
 			fname = g.Function.Name
 		}
-		fmt.Printf("Goroutine %d - %s:%d %s\n", g.ID, g.File, g.Line, fname)
+		fmt.Printf("Goroutine %d - %s:%d %s (%d)\n", g.ID, g.File, g.Line, fname, g.PC)
```
Comment (derekparker) [inline terminal/command.go]: Change to (%#v) for printing PC.

**Comment by derekparker** [comment]
The tests pass on OSX but are failing for me on Linux. Do you have access to a Linux box you can use to fix them?

**Comment by aarzilli** [comment]
How are the tests failing? I'm on linux/go1.4.2 and I don't see any errors.

**Comment by derekparker** [comment]
This is the error I'm seeing:

```
vagrant@vagrant-ubuntu-trusty-64:~/go/src/github.com/derekparker/delve/proc$ go version
go version go1.4.2 linux/amd64
vagrant@vagrant-ubuntu-trusty-64:~/go/src/github.com/derekparker/delve/proc$ uname -a
Linux vagrant-ubuntu-trusty-64 3.13.0-55-generic #92-Ubuntu SMP Sun Jun 14 18:32:20 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
vagrant@vagrant-ubuntu-trusty-64:~/go/src/github.com/derekparker/delve/proc$ go test -run Stack
--- FAIL: TestStacktraceGoroutine-2 (0.10s)
        proc_test.go:504: Goroutine stacks not found ([{-1 %!d(string=runtime.goparkunlock)} {-1 %!d(string=runtime.chansend)} {-1 %!d(string=runtime.chansend1)} {6 %!d(string=main.agoroutine)}])
FAIL
exit status 1
FAIL    github.com/derekparker/delve/proc       0.161s
```

**Comment by aarzilli** [comment]
bad test design, calling Gosched fixes it but of course we can not really guarantee in general the goroutines make progress. Bizarre that with GOMAXPROCS = 4 it worked deterministically.

**Comment by derekparker** [comment]
Good catch. I rebased everything into 1 commit and updated the commit message, but it's merged in now.

Thanks for the patch!

**Pull request closed without merging**